src/rulesets/Base/distributed.jl

```diff
@@ -0,0 +1,61 @@
+#####
+##### `pmap`
+#####
+
+# Now that there is a backwards rule for zip (albeit only in Zygote),
+# it should be fine to deal with only a single collection X
+function rrule(config::RuleConfig{>:HasReverseMode}, ::typeof(pmap), f, p::AbstractWorkerPool, X; kwargs...)
+    project_X = ProjectTo(X)
+
+    darr = dfill([], (nworkers(p) + 1,), vcat(myid(), workers(p))) # Include own proc to handle empty worker pool
+
+    function forw(x)
+        y, back = rrule_via_ad(config, f, x)
+        push!(darr[:L][1], back)
+        return y, myid(), length(darr[:L][1])
+    end
+
+    ys_IDs_indices = pmap(forw, p, X; kwargs...)
+    ys = getindex.(ys_IDs_indices, 1) # the primal values
+    IDs = getindex.(ys_IDs_indices, 2) # remember which processors handled which elements of X
+    indices = getindex.(ys_IDs_indices, 3) # remember the index of the pullback in the array on each processor
```

Comment:
What types are these? 

Is it worth making arrays from the array of tuples, not just directly reading them out as needed?

But more, since `forw` already mutates something as it goes along, to store the pullbacks, is it necessary to also create an array of tuples? Or could the details similarly be stored as it goes?